xarray/core/missing.py

```diff
@@ -631,8 +632,14 @@ def interp(var, indexes_coords, method: InterpOptions, **kwargs):
         return var.copy()
 
     result = var
-    # decompose the interpolation into a succession of independent interpolation
-    for indep_indexes_coords in decompose_interp(indexes_coords):
+
+    if method in ["linear", "nearest"]:
```

Comment:
The test seems to suggest that `"pchip"` and `"slinear"` are acceptable too

Reply:
I'm getting an error with nans in intermediate interps using pchip so I'm going to leave it out of the decompose list